### PR TITLE
Fixed display of column values in downloaded tsv

### PR DESCRIPTION
**Pull request by jtquach1**
Fix cBioPortal/cbioportal#7911

Describe changes proposed in this pull request:
- Added download function for Cytoband
- Removed columns and their headers that don't have a download function defined

## Before
image
[table.tsv.txt](https://github.com/cBioPortal/cbioportal-frontend/files/5513249/table.tsv.txt)

## After
image
[table (15).tsv.txt](https://github.com/cBioPortal/cbioportal-frontend/files/5513250/table.15.tsv.txt)

### CODE REVIEW

**Review by alisman — APPROVED** [review]

File: src/shared/components/lazyMobXTable/LazyMobXTable.tsx [hunk 1]
```diff
@@ -378,7 +378,29 @@ export class LazyMobXTableStore<T> {
                 }
             );
         });
-        return tableDownloadData;
+
```
Comment (Luke-Sikina) [inline src/shared/components/lazyMobXTable/LazyMobXTable.tsx:381]: I don't like that we process all the columns and _then_ filter out columns without the download function. Futhermore, you're just filtering out empty columns, which could theoretically be legit data. 

Can you spend some time seeing if you can instead filter out columns before we create tableDownloadData?

File: src/pages/patientView/copyNumberAlterations/CopyNumberTableWrapper.tsx [hunk 2]
```diff
@@ -242,6 +242,8 @@ export default class CopyNumberTableWrapper extends React.Component<
             ),
             sortBy: (d: DiscreteCopyNumberData[]) =>
                 this.getCytobandForGene(d[0].gene.hugoGeneSymbol),
+            download: (d: DiscreteCopyNumberData[]) =>
```
Comment (Luke-Sikina) [inline src/pages/patientView/copyNumberAlterations/CopyNumberTableWrapper.tsx:245]: Seems good to me

**Review by Luke-Sikina — CHANGES_REQUESTED** [review]

**Comment by jtquach1** [comment]
Current output (same):
image

[table (3).tsv.txt](https://github.com/cBioPortal/cbioportal-frontend/files/5527391/table.3.tsv.txt)

File: src/shared/components/lazyMobXTable/LazyMobXTable.tsx (hunk 1 above)
Comment (jtquach1) [reply src/shared/components/lazyMobXTable/LazyMobXTable.tsx]: That's true- I just realized it's inefficient to filter at the very end (and a specified download function could return the empty string). I just saw that the header and an empty string for the download object rows are displayed for columns without specified download functions.

**Review by Luke-Sikina — CHANGES_REQUESTED** [review]

File: src/shared/components/lazyMobXTable/LazyMobXTable.spec.tsx [hunk 3]
```diff
@@ -398,6 +398,12 @@ describe('LazyMobXTable', () => {
         clock.uninstall();
     });
 
+    describe('downloadData', () => {
```
Comment (alisman) [inline src/shared/components/lazyMobXTable/LazyMobXTable.spec.tsx:401]: probably you're working on this

Comment (jtquach1) [reply src/shared/components/lazyMobXTable/LazyMobXTable.spec.tsx:401]: that's correct, I'm in the process of mocking up a props.columns and props.data to pass into a LazyMobXTable constructor call

File: src/shared/components/lazyMobXTable/LazyMobXTable.tsx [hunk 4]
```diff
@@ -364,7 +357,7 @@ export class LazyMobXTableStore<T> {
             //rowDownloadData is list of lists, containing all the elements per column.
             //processedRowsDownloadData becomes the transposed of rowDownloadData.
             let processedRowsDownloadData = rowDownloadData[0].map(function(
-                row: string,
+                _row: string,
```
Comment (jtquach1) [inline src/shared/components/lazyMobXTable/LazyMobXTable.tsx:360]: VS Code suggested prefixing unused variables with an underscore

Comment (Luke-Sikina) [reply src/shared/components/lazyMobXTable/LazyMobXTable.tsx:360]: Let's not do that. I guess that's vscode's default naming scheme for unused variables, but that clashes with how we use underscore prefixed variables in our stores. Just leave it as `row`.

File: src/shared/components/lazyMobXTable/LazyMobXTable.spec.tsx (hunk 3 above)
Comment (Luke-Sikina) [reply src/shared/components/lazyMobXTable/LazyMobXTable.spec.tsx:402]: This looks good, nice job. Given the amount of boilerplate we'll have to add if we do multiple tests, I think this is good enough. Thank you for adding the test.

File: src/shared/components/lazyMobXTable/LazyMobXTable.spec.tsx [hunk 5]
```diff
@@ -398,6 +399,25 @@ describe('LazyMobXTable', () => {
...
+        it.only('only displays columns with defined download functions', () => {
```
Comment (Luke-Sikina) [inline src/shared/components/lazyMobXTable/LazyMobXTable.spec.tsx:403]: Oop, you left an `only`
```suggestion
        it('only displays columns with defined download functions', () => {
```

**Review by Luke-Sikina — APPROVED** [review]
Tiny changes. This looks really good, nice job.

File: src/shared/components/lazyMobXTable/LazyMobXTable.tsx [hunk 6]
```diff
@@ -787,7 +780,7 @@ export default class LazyMobXTable<T> extends React.Component<
                 // populate the cache instances with all available data for the lazy loaded columns
                 this.store.downloadDataFetcher
                     .fetchAndCacheAllLazyData()
-                    .then(allLazyData => {
+                    .then(_allLazyData => {
```
Comment (Luke-Sikina) [inline src/shared/components/lazyMobXTable/LazyMobXTable.tsx:783]: couldn't you just replace this one with `()`?

File: src/shared/components/lazyMobXTable/LazyMobXTable.tsx [hunk 7]
```diff
@@ -872,7 +865,7 @@ export default class LazyMobXTable<T> extends React.Component<
         );
         this.pageToHighlightReaction = reaction(
             () => this.store.firstHighlightedRowIndex,
-            (index: number) => {
+            (_index: number) => {
```
Comment (Luke-Sikina) [inline src/shared/components/lazyMobXTable/LazyMobXTable.tsx:868]: Same, doesn't `()` work?

File: src/shared/components/lazyMobXTable/LazyMobXTable.tsx (hunk 6 above)
Comment (jtquach1) [reply src/shared/components/lazyMobXTable/LazyMobXTable.tsx:783]: yes, it looks like passing in a no-arg lambda works (no type mismatch)

**Review by Luke-Sikina — APPROVED** [review]

**Pull request merged**